robot-code/cpp/CommsManager.h

```diff
@@ -38,15 +44,103 @@ class CommsManager
                      std::string robot_name):
                     p_ssid{ssid},
-                    p_pwd{password},
-                    p_robot_name{robot_name}{};
-        GameState getGameState() const{return };
-};
\ No newline at end of file
+                    p_pwd{password}
+                    {
+                        if(robot_name.length());
+                        g_robot_name = robot_name;
+                    };
+        GameState getGameState() const{return p_game_state;};
+        void Initialize();
+};
+
+//static allocations
+JoystickBuffer CommsManager::g_js_buffer{0};
+ManagerBuffer CommsManager::g_manger_buffer{0};
+std::string CommsManager::g_robot_name;
+uint8_t CommsManager::g_command_hint{0};
+
+void CommsManager::UdpCallback(AsyncUDPPacket packet)
+{
+    if(packet.length() == sizeof(JoystickBuffer))
+    {
+        if(memcmp(packet.data(), g_robot_name.c_str(),g_robot_name.length()))
+        {
+            memcpy(&g_js_buffer, packet.data(), sizeof(g_js_buffer));
+        }
+    }
+
+    if(packet.length() == sizeof(ManagerBuffer))
+    {
+        bool in_match = false;
+        auto data_peek = reinterpret_cast<ManagerBuffer *>(packet.data());
+ 
+        switch (g_command_hint)
+        {
+        case 0:
+            {
+                std::array data = {data_peek->robot_1, // There might be a faster way to make these references, if performace limits are hit take a look here
+                        data_peek->robot_2,
+                        data_peek->robot_3,
+                        data_peek->robot_4}; 
+
+                for(int i = 0; i < data.size(); i++)
+                {
+                    if(memcmp(data[i],g_robot_name.c_str(), g_robot_name.length()))
+                    {
+                        g_command_hint = i+1; // The case statement is 1 indexed since the zero case is for no hint
+                        in_match = true;
+                    }
+                }
+            }        
+            break;
+        case 1:
+            if(memcmp(data_peek->robot_1,g_robot_name.c_str(), g_robot_name.length())) in_match = true;
+            break;
+        case 2:
+            if(memcmp(data_peek->robot_2,g_robot_name.c_str(), g_robot_name.length())) in_match = true;
+            break;
+        case 3:
+            if(memcmp(data_peek->robot_3,g_robot_name.c_str(), g_robot_name.length())) in_match = true;
+            break;
+        case 4:
+            if(memcmp(data_peek->robot_4,g_robot_name.c_str(), g_robot_name.length())) in_match = true;
+            break;
+        }
+
+        if(in_match)
+        {
+            g_manger_buffer.command_word = packet.data()[0];
+        }
+    }
+}
+
+void CommsManager::Initialize()
+{
+    WiFi.mode(WIFI_STA);
+    WiFi.begin(p_ssid.c_str(), p_pwd.c_str());
+    if (WiFi.waitForConnectResult() != WL_CONNECTED)
+    {
+        for(;;);//halt and catch fire for now, maybe handle it latter. At least blink aggressivly. 
```

Comment:
This doesn't try a reconnect, should loop but try for a connection after some delay